sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -98,6 +98,42 @@ case class DataSourceResolution(
 
     case DropViewStatement(AsTableIdentifier(tableName), ifExists) =>
       DropTableCommand(tableName, ifExists, isView = true, purge = false)
+
+    case AlterTableSetPropertiesStatement(AsTableIdentifier(table), properties) =>
+      AlterTableSetPropertiesCommand(table, properties, isView = false)
+
+    case AlterViewSetPropertiesStatement(AsTableIdentifier(table), properties) =>
+      AlterTableSetPropertiesCommand(table, properties, isView = true)
+
+    case AlterTableUnsetPropertiesStatement(AsTableIdentifier(table), propertyKeys, ifExists) =>
+      AlterTableUnsetPropertiesCommand(table, propertyKeys, ifExists, isView = false)
+
+    case AlterViewUnsetPropertiesStatement(AsTableIdentifier(table), propertyKeys, ifExists) =>
+      AlterTableUnsetPropertiesCommand(table, propertyKeys, ifExists, isView = true)
+
+    case AlterTableSetLocationStatement(AsTableIdentifier(table), newLocation) =>
+      AlterTableSetLocationCommand(table, None, newLocation)
+
+    case AlterTableAddColumnsStatement(AsTableIdentifier(table), newColumns)
+        if newColumns.forall(_.name.size == 1) =>
+      // only top-level adds are supported using AlterTableAddColumnsCommand
+      val newFields = newColumns.map { newCol =>
+        val builder = new MetadataBuilder
+        newCol.comment.foreach(builder.putString("comment", _))
+
+        val cleanedDataType = HiveStringType.replaceCharType(newCol.dataType)
+        if (newCol.dataType != cleanedDataType) {
+          builder.putString(HIVE_TYPE_STRING, newCol.dataType.catalogString)
+        }
+
+        StructField(
+          newCol.name.head,
+          cleanedDataType,
+          nullable = true,
+          builder.build())
+      }
```

Comment:
Could you make a function for line 120~134 because it seems that we need to reuse this logic for `AlterTableAlterColumnStatement` very soon?

Reply:
Done.